### PR TITLE
Drupal 8 Port.

### PR DESCRIPTION
**Pull request by willtp87**
d8 port. Requires https://github.com/discoverygarden/islandora_paged_content/pull/1

### CODE REVIEW

File: config/install/islandora_book.settings.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+islandora_book_ingest_derivatives: ['pdf', 'image', 'ocr']
```
Comment (adam-vessey) [inline config/install/islandora_book.settings.yml:1]: Not really... _wrong_... just weird to see a list in YAML inline... shrug?

File: islandora_book.routing.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+islandora_book.admin:
+  path: '/admin/config/islandora/solution_pack_config/book'
+  defaults:
+    _title: 'Book Solution Pack'
+    _form: '\Drupal\islandora_book\Form\Admin'
+  requirements:
+    _permission: 'administer site configuration'
+islandora_book.pages:
+  path: 'islandora/object/{object}/pages'
```
Comment (adam-vessey) [inline islandora_book.routing.yml]: Not sure if it really effects anything, but this is the only path without an initial slash.

File: theme/theme.inc [hunk 3]
```diff
@@ -77,40 +64,24 @@ function template_preprocess_islandora_book_page(array &$variables) {
...
     // Get token to allow access to XACML protected datastreams.
     // Always use token authentication in case there is a global policy.
-    module_load_include('inc', 'islandora', 'includes/authtokens');
-    $token = islandora_get_object_token($object->id, 'JP2', 2);
-    $jp2_url = url("islandora/object/{$object->id}/datastream/JP2/view", array(
-      'absolute' => TRUE,
-      'query' => array('token' => $token),
-    ));
+    $token = TokenAuth::getObjectToken($object->id, 'JP2', 2);
     $params['token'] = $token;
```
Comment (adam-vessey) [inline theme/theme.inc:68]: Probably really don't care now... just a thought, but: Could possibly do make this lazily evaluate... only generate a token if used as a string, kind of thing...

... yeah, crazy ramblings, disregard.

File: theme/theme.inc [hunk 4]
```diff
@@ -77,40 +64,24 @@ function template_preprocess_islandora_book_page(array &$variables) {
...
- * Processes variables for islandora_book_page templates.
- *
- * Default template: islandora-book-page.tpl.php.
- *
- * @param array $variables
- *   An associative array containing:
- *   - object: An AbstractObject for which to generate the display.
- *   - viewer_id: The "ID" of the viewer to use.
- *   - viewer_params: An (associative) array of parameters to pass to the
- *     viewer.
```
Comment (adam-vessey) [inline theme/theme.inc:111]: Should these variables make their way up into the merged function doc?

File: theme/theme.inc [hunk 5]
```diff
@@ -19,37 +23,20 @@ function template_preprocess_islandora_book_book(array &$variables) {
...
- * Processes variables for islandora_book_book templates.
- *
- * Default template: islandora-book-book.tpl.php.
- *
- * @param array $variables
- *   An associative array containing:
- *   - object: An AbstractObject for which to generate the display.
- *   - viewer_id: The "ID" of the viewer to use.
- *   - viewer_params: An (associative) array of parameters to pass to the
- *     viewer.
```
Comment (adam-vessey) [inline theme/theme.inc:50]: Should these variables make their way up into the merged function doc?

File: theme/theme.inc (hunk 4 above)
Comment (adam-vessey) [reply theme/theme.inc:111]: Also, said function doc is still referring to a `.tpl.php`... is now twig, yes?

File: theme/theme.inc (hunk 5 above)
Comment (adam-vessey) [reply theme/theme.inc:50]: Also, said function doc is still referring to a `.tpl.php`... is now twig, yes?

File: islandora_book.routing.yml (hunk 2 above)
Comment (adam-vessey) [reply islandora_book.routing.yml]: [Docs indicate it should be present](https://www.drupal.org/docs/8/api/routing-system/structure-of-routes):

> **path (required)**: The URL to the route, _with a leading forward slash_

**Review by adam-vessey — CHANGES_REQUESTED** [review]
Really, just the path in the `routing.yml`... _could_ preserve and adapt some of the function comments in the theme stuff but... shrug?

File: theme/theme.inc (hunk 3 above)
Comment (willtp87) [reply theme/theme.inc:68]: Disregarding.

File: config/install/islandora_book.settings.yml (hunk 1 above)
Comment (willtp87) [reply config/install/islandora_book.settings.yml:1]: shrug

File: theme/theme.inc (hunk 4 above)
Comment (willtp87) [reply theme/theme.inc:111]: They don't really apply as params to the preprocess. I think the proper thing would be updating the templates and that seems like an out of scope improvement.

File: theme/theme.inc (hunk 5 above)
Comment (willtp87) [reply theme/theme.inc:50]: https://github.com/discoverygarden/islandora_book/pull/1/files/da93727aea3bf9d6922cd0277cc8092c4b7449bd#r249935498

**Review by adam-vessey — APPROVED** [review]

**Comment by adam-vessey** [comment]
Couple of things, but is fine:
* Ran into https://jira.duraspace.org/browse/ISLANDORA-2358
* Also, ran into a funky issue with the heights/widths of the images in the swaptable business; however, appears to be due to the theme being used (seven, on the admin page)... Given it's external stuff... fine as-is, for now. Given the most books have relatively similar dimensions... yeah, it's fine.

**Pull request merged**